### DISCUSSION TITLE
Add collaborative writing and calendar.

### DISCUSSION THREAD

**Discussion by albanobattistella** [learningequality/kolibri#8116] [Q&A]
Hello Team!

to be complete, Kolibri needs a collaborative writing sheet and calendar.
I would recommend etherpad, very light, effective, and safe.
For the calendar, implement it in kolibri also to give scheduled lessons to students by teachers.
The students themselves will find all the tasks to be done on the calendar.
V/R,
Albano

**Comment by jonboiser** [comment]
Thanks for the suggestions. We have many improvements to Kolibri in our roadmap, and like to use our Github Issues list for bugs and tasks to be done in the near term.

**Comment by rtibbles** [comment]
One thing to note is that etherpad-lite is implemented with a nodejs backend, so would require considerable work to make it portably integrated with Kolibri.

On the calendar side, one reason we have resisted implementing this kind of feature so far has been that many devices that we have seen Kolibri installed on have suffered from catastrophic internal clock failures, which make any sort of calendar feature very unreliable. If you have thoughts around mitigating these kinds of issues for our users, that would be very useful!

**Reply by celestial-30** [reply]
Could you use a system that averages time between the connected instances of kolibri? this may not work for dates if all instances are down (due to power outages for instance) but if its simply that the computer running one instance is damaged, it could ask for updated time if it detects that it is out of sync with the other instances?

**Comment by albanobattistella** [comment]
Hi,

I have been trying to ask for features in the forum for a long time but have not gotten any answers since last year.
that's why I posted my comments here.
about eterphad-lite and the calendar, it was an idea for the services to be implemented because many teachers require it.
I did all the translation of kolibri in Italian ... and now I'm going to finish the documentation.
I am happy to help you with other things as well.
Thanks for your answers.
Albano

________________________________
Da: Jonathan Boiser ***@***.***>
Inviato: mercoledì 26 maggio 2021 17:44
A: learningequality/kolibri ***@***.***>
Cc: albanobattistella ***@***.***>; Author ***@***.***>
Oggetto: Re: [learningequality/kolibri] Add collaborative writing and calendar. (#8104)


Thanks for the suggestions. We have many improvements to Kolibri in our roadmap, and like to use our Github Issues list for bugs and tasks to be done in the near term.

**Comment by jonboiser** [comment]
Hi Albano,

I'm sorry about your experience on the forum. We must have lost track of the discussion. [Link](https://community.learningequality.org/t/future-request-for-kolibri/1576)

I do not think it will be possible to add these features in Kolibri any time soon, but I do think it is possible to use something like etherpad and a calendar _alongside_ Kolibri through the use of linking. For example, each student lesson has a unique link that can be placed in calendar app. I will convert this issue to a discussion if you want to continue it here.
